thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/HoltWintersDetector.java

```diff
@@ -112,13 +112,25 @@ public void init(HoltWintersDetectorSpec spec, InputDataFetcher dataFetcher) {
     this.pattern = spec.getPattern();
     this.smoothing = spec.getSmoothing();
     this.sensitivity = spec.getSensitivity();
+    this.monitoringGranularity = spec.getMonitoringGranularity();
+
+    if (this.monitoringGranularity.equals("1_MONTHS")) {
+      this.timeGranularity = MetricSlice.NATIVE_GRANULARITY;
```

Comment:
What is the rationale behind this logic?

Reply:
Pinot doesn't support monthly aggregation from the query. Monthly detection is only doable if only the dataset itself is marked as daily granularity but only have one data point per month. So for monthly detection, when fetching data should still use native granularity, but the algorithm needs to be aware that it's a monthly granularity data and fetches the right training data to construct the correct baseline. 

@tangdian I'm not sure this will work for the monthly dataset. looks like you are still retrieving 60 days' training data. Then it's likely that there only 1 data point in there. Since this is not required, you can throw a NotImpelemented exception in this case if it's hard to make it work.

Reply:
According to Jihao, monthly data from pinot will only have monthly data points. i.e. It will not do aggregation. So we will set timeGranularity to NATIVE_GRANULARITY.

Reply:
> Pinot doesn't support monthly aggregation from the query. Monthly detection is only doable if only the dataset itself is marked as daily granularity but only have one data point per month. So for monthly detection, when fetching data should still use native granularity, but the algorithm needs to be aware that it's a monthly granularity data and fetches the right training data to construct the correct baseline.
> 
> @tangdian I'm not sure this will work for the monthly dataset. looks like you are still retrieving 60 days' training data. Then it's likely that there only 1 data point in there. Since this is not required, you can throw a NotImpelemented exception in this case if it's hard to make it work.

I have updated the code to support 1_MONTHS data